### PR TITLE
docs: chemical system theory

### PR DESCRIPTION
**Pull request by richardjgowers**
fixes issue #759

### CODE REVIEW

File: docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst [hunk 1]
```diff
@@ -1,4 +1,42 @@
...
+Splitting the total system into components serves two purposes:
+
+* the thermodynamic cycles can be easily understood by comparing the differences in Components
+* ``Protocol`` \s can know to treat different components differently, for example applying different force fields
+
+See Also
+--------
+
+* To see how to construct ``ChemicalSystem`` \s from your files, see :ref:`the cookbook entry on loading molecules <Loading Molecules>`
+* For details of what thermodynamic cycles to construct, consult the pages for each specific ``Protocol``
```
Comment (IAlibay) [inline docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst:42]: "How to arrange ChemicalSystems to define a the alchemical transformations necessary for a given thermodynamic cycle"?

Also possibly link `Protocol` here to  `_userguide_protocols`?

File: docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst [hunk 2]
```diff
@@ -1,4 +1,42 @@
...
+When thinking about computing estimates in the free energy differences between different states,
+we need an object model to represent the chemistry we are considering.
+In ``openfe`` a :class:`.ChemicalSystem` is used to capture this information,
+and represents the chemical models that are present in each end state.
+
+A ``ChemicalSystem`` **does** include information, where present, on:
+
+* exact protonation state of both the protein and ligand
+* the exact pose of a ligand or conformation of a protein
+* the presence of any crystallographic waters
```
Comment (IAlibay) [inline docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst:15]: Would it also be worth talkinga bout solvation details, e.g. the necessary details to put the system in a box of water?

**Review by IAlibay — COMMENTED** [review]
Might be worth another set of eyes on this one, not sure if I fully understood the aim of this page.

File: docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst [hunk 3]
```diff
@@ -1,4 +1,71 @@
...
 
-issue #759
\ No newline at end of file
+ChemicalSystems, Components and Thermodynamic Cycles
+====================================================
+
+Chemical Systems
+----------------
+
+In order to define the input systems to a Protocol, so as the end states of an alchemical transformation,
```
Comment (IAlibay) [inline docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst:9]: Apologies, the "so as the end states of the alchemical transformation" part of this sentence doesn't really make sense to me - I think I know what you're trying to say, but the sentence may need restructuring?

File: docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst [hunk 4]
```diff
@@ -1,4 +1,71 @@
...
+
+It **does not** include any information on:
+
+* forcefield applied to any component, including details on water model or virtual particles
+* thermodynamic conditions, i.e. temperature and pressure
+
+Components
+----------
+
+A ``ChemicalSystem`` is composed of many ``Component`` objects,
```
Comment (IAlibay) [inline docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst:29]:
```suggestion
A :class:`.ChemicalSystem` is composed of many :class:`.Component` objects,
```

File: docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst [hunk 5]
```diff
@@ -1,4 +1,71 @@
...
+  ligand1_complex = openfe.ChemicalSystem(components={'ligand': ligand1, 'protein': protein, 'solvent': solvent})
+  ligand2_complex = openfe.ChemicalSystem(components={'ligand': ligand2, 'protein': protein, 'solvent': solvent})
+  ligand1_solvent = openfe.ChemicalSystem(components={'ligand': ligand1, 'solvent': solvent})
+  ligand2_solvent = openfe.ChemicalSystem(components={'ligand': ligand2, 'solvent': solvent})
+
+
+See Also
+--------
+
+* To see how to construct ``ChemicalSystem`` \s from your files, see :ref:`the cookbook entry on loading molecules <Loading Molecules>`
```
Comment (IAlibay) [inline docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst:70]:
```suggestion
* To see how to construct a :class:`.ChemicalSystem` \s from your files, see :ref:`the cookbook entry on loading molecules <Loading Molecules>`
```

File: docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst [hunk 6]
```diff
@@ -1,4 +1,71 @@
...
+::
+
+  import openfe
+
+  ligand1 = openfe.SmallMoleculeComponent.from_sdf_file('./ligand1.sdf')
+  ligand2 = openfe.SmallMoleculeComponent.from_sdf_file('./ligand2.sdf')
+  protein = openfe.ProteinComponent.from_pdb_file('./protein.pdb')
+  solvent = openfe.SolventComponent()
+
+  ligand1_complex = openfe.ChemicalSystem(components={'ligand': ligand1, 'protein': protein, 'solvent': solvent})
```
Comment (IAlibay) [inline docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst:61]: Maybe worth adding comments to explain what each of the four chemicalsystems represent?

File: docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst [hunk 7]
```diff
@@ -1,4 +1,71 @@
...
+* alchemical transformations can be easily understood by comparing the differences in Components
+* ``Protocol`` \s can know to treat different components differently, for example applying different force fields
+
+Thermodynamic Cycles
+--------------------
+
+With a language to express chemical systems piecewise, we can now also construct thermodynamic cycles based on these.
+For example to  construct the classic relative binding free energy cycle, we will need four components, two ligands,
+a protein, and a solvent.  These four ingredients can then be combined into the four point on the thermodynamic cycle
+that we wish to sample:
```
Comment (IAlibay) [inline docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst:48]: I would suggest a note here that says that the required chemicalsystem inputs, and the exact interpretation of a thermodynamic cycle, may vary from Protocol to Protocol.

File: docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst [hunk 8]
```diff
@@ -1,6 +1,84 @@
...
+which correspond as the end states of an alchemical transformation,
+we need an object model to represent their chemical composition.
+In ``openfe`` a :class:`.ChemicalSystem` is used to capture this information,
+and represents the chemical models that are present in each end state.
+
+A :class:`.ChemicalSystem` **does** include information, where present, on:
+
+* exact atomic information (including protonation state) of protein, ligands, co-factors, and any crystallographic
+  waters
+* atomic positions of all explicitly  ligand or conformation of a protein
```
Comment (hannahbaumann) [inline docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst:19]: I don't understand what this bullet point means. Could you maybe rephrase?

File: docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst [hunk 9]
```diff
@@ -1,4 +1,42 @@
...
+
+A ``ChemicalSystem`` is composed of many ``Component`` objects,
+each representing a single ''piece'' of the overall system.
+Examples of components include:
+
+* :class:`.ProteinComponent`
+* :class:`.SmallMoleculeComponent`
+* :class:`.SolventComponent`
+
+Splitting the total system into components serves two purposes:
```
[the remark this answers is not in the document]
Comment (hannahbaumann) [reply docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst:38]: Has this been addressed?

**Review by hannahbaumann — CHANGES_REQUESTED** [review]

File: docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst [hunk 10]
```diff
@@ -1,6 +1,85 @@
...
+which correspond as the end states of an alchemical transformation,
+we need an object model to represent their chemical composition.
+In ``openfe`` a :class:`.ChemicalSystem` is used to capture this information,
+and represents the chemical models that are present in each end state.
+
+A :class:`.ChemicalSystem` **does** include information, where present, on:
+
+* exact atomic information (including protonation state) of protein, ligands, co-factors, and any crystallographic
+  waters
+* atomic positions of all explicitly components such as ligands or proteins
```
Comment (hannahbaumann) [inline docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst:19]: What is an "explicitly component"? Maybe explicitly defined components?

**Review by hannahbaumann — CHANGES_REQUESTED** [review]

File: docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst (hunk 10 above)
Comment (richardjgowers) [reply docs/guide/setup/chemical_systems_and_thermodynamic_cycles.rst]: yep fixed

**Review by hannahbaumann — APPROVED** [review]
